include/sys/txg.h

```diff
@@ -66,6 +66,21 @@ typedef struct txg_list {
 	txg_node_t	*tl_head[TXG_SIZE];
 } txg_list_t;
 
+/*
+ * Wait flags for txg_wait_synced_flags(). By default (TXG_WAIT_NONE), it will
+ * wait until the wanted txg is reached, or block forever. Additional flags
+ * indicate other conditions that the caller is interested in, that will cause
+ * the wait to break and return an error code describing the condition.
+ */
+/*
+ * No special flags. Guaranteed to block forever or return 0.
+ */
+#define	TXG_WAIT_NONE		(0)
+/*
+ * If a signal arrives while waiting, abort and return EINTR.
+ */
+#define	TXG_WAIT_SIGNAL		(1 << 0)
```

Comment:
Do we prefer defines to enum?  In https://github.com/openzfs/zfs/pull/17218 use of enum allowed me to catch some improper use cases.

Reply:
Yeah, nice. I generally prefer enums because debuggers can see them, but I've just never thought much about whether we have a strong preference. Guess that starts today: changed!